tests/roots/test-linkcheck-connection-error-on-http-head/index.rst

```diff
@@ -0,0 +1,3 @@
+This tests bug #9306.
+
+This is `a link to the US Patent Website <https://patft.uspto.gov/netacgi/nph-Parser?Sect1=PTO2&Sect2=HITOFF&p=1&u=%2Fnetahtml%2FPTO%2Fsearch-bool.html&r=1&f=G&l=50&co1=AND&d=PTXT&s1=7840660&OS=7840660&RS=7840660>`_
```

Comment:
I wonder if the behavior can be emulated with the `http_server` context manager?
My concern is that if this site changes behavior, the test will stop being useful.
That would be much faster (runs over localhost).

Reply:
That's a good idea. I will attempt to replace the test with one that doesn't depend on an external service.